### PR TITLE
feat: add Pulsar bindings to the specification.

### PR DESCRIPTION
**Pull request by VisualBean**
title: "Add pulsar bindings and protocol to the specification"
---

**Related issue(s):**

[Pulsar binding PR](https://github.com/asyncapi/bindings/pull/173)
[Json schema update PR](https://github.com/asyncapi/spec-json-schemas/pull/295)

The pulsar binding was finalized in the Bindings repository.
This PR adds the new pulsar binding objects, to the Server and Channel bindings list.
Along with adding "pulsar" to the protocol list.

### CODE REVIEW

**Review by fmvilas — CHANGES_REQUESTED** [review]
So far looks good to me but would you mind adding the operation and message bindings too? Even if we don't have any of those in the Pulsar binding yet but just so we have it here as a placeholder.

**Comment by VisualBean** [comment]
> So far looks good to me but would you mind adding the operation and message bindings too? Even if we don't have any of those in the Pulsar binding yet but just so we have it here as a placeholder.

I have added these.

**Review by fmvilas — DISMISSED** [review]
LGTM 👍

**Review by char0n — APPROVED** [review]
LGTM :+1:

**Comment by char0n** [comment]
One small comment: do we want to update following section https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#protocol and add Google Pub/Sub and Pulsar there as well?

> Example protocols include, but are not limited to, AMQP, HTTP, JMS, Kafka, Anypoint MQ, MQTT, Solace, STOMP, Mercure, WebSocket, Google Cloud Pub/Sub, Pulsar.

**Comment by VisualBean** [comment]
> One small comment: do we want to update following section https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#protocol and add Google Pub/Sub and Pulsar there as well?
> 
> > Example protocols include, but are not limited to, AMQP, HTTP, JMS, Kafka, Anypoint MQ, MQTT, Solace, STOMP, Mercure, WebSocket, Google Cloud Pub/Sub, Pulsar.

Done

File: spec/asyncapi.md
Comment (VisualBean) [inline spec/asyncapi.md:108]:
```suggestion
A protocol is the mechanism (wireline protocol or API) by which [messages](#definitionsMessage) are exchanged between the application and the [channel](#definitionsChannel). Example protocols include, but are not limited to, AMQP, HTTP, JMS, Kafka, Anypoint MQ, MQTT, Solace, STOMP, Mercure, WebSocket, Google Pub/Sub, Pulsar.  
```

**Review by char0n — APPROVED** [review]

**Review by fmvilas — DISMISSED** [review]

**Comment by fmvilas** [comment]
> One small comment: do we want to update following section https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#protocol and add Google Pub/Sub and Pulsar there as well?
> 
> > Example protocols include, but are not limited to, AMQP, HTTP, JMS, Kafka, Anypoint MQ, MQTT, Solace, STOMP, Mercure, WebSocket, Google Cloud Pub/Sub, Pulsar.

Just one small thing: this paragraph will be removed in v3.0.0. It's not intended to be an exhaustive list of protocols and very often I see folks getting confused thinking it actually is. Even though it says "not limited to" 😅

**Comment by VisualBean** [comment]
Yeah, i completely agree, removing it, to lessen confusion makes sense. 
Also it's not really an actual protocols list per say, as some of these would use their own format anyway like `http` or `pulsar+ssl`.

**Comment by derberg** [comment]
if this is planned for next release, target branch should not be `master` but `next-spec`

**Review by derberg — DISMISSED** [review]
I set label to RFC 3 as it was already approved by 3 codeowners and binding before was also already approved and seen by others

also we have https://github.com/asyncapi/spec-json-schemas/pull/295

the only thing blocking is really the fact that these 2 PRs point to `master` branches and should to `next-spec`

**Comment by fmvilas** [comment]
Changed the base branch but got conflicts now. @VisualBean I can't fix the conflicts for you because I don't have permission on your repo/branch. Please go ahead and fix them and that should be all.

**Comment by VisualBean** [comment]
> Changed the base branch but got conflicts now. @VisualBean I can't fix the conflicts for you because I don't have permission on your repo/branch. Please go ahead and fix them and that should be all.

Awesome, thanks. I'll get them worked out

**Comment by derberg** [comment]
@VisualBean due to update in `next-spec` branch there are some conflicts that you need to resolve, sorry

can you also change target in https://github.com/asyncapi/spec-json-schemas/pull/295 please, point to `next-spec` as well

**Review by fmvilas — APPROVED** [review]

**Comment by AceTheCreator** [comment]
@fmvilas can you merge this PR?

**Review by derberg — APPROVED** [review]

**Review by smoya — APPROVED** [review]
LGTM!

**Review by dalelane — APPROVED** [review]

**Comment by derberg** [comment]
Recent comments about the release from the bot were added by mistake. More details in https://github.com/asyncapi/spec/issues/899

**Pull request merged**
